lib/spack/spack/environment.py

```diff
@@ -1944,7 +1944,10 @@ def _update_and_write_manifest(self, raw_yaml_dict, yaml_dict):
         written = os.path.exists(self.manifest_path)
         if changed or not written:
             self.raw_yaml = copy.deepcopy(self.yaml)
-            with fs.write_tmp_and_move(self.manifest_path) as f:
+            real_manifest_path = self.manifest_path
+            if os.path.islink(self.manifest_path):
+                real_manifest_path = os.readlink(self.manifest_path)
+            with fs.write_tmp_and_move(real_manifest_path) as f:
```

Comment:
This does not work with chained symlinks, use realpath instead:
```suggestion
            with fs.write_tmp_and_move(os.path.realpath(self.manifest_path)) as f:
```

This doesn't work for chained symlinks.